pulp_docker/tests/functional/api/test_publish.py

```diff
@@ -77,21 +68,13 @@ def test_all(self):
         non_latest = choice(version_hrefs[:-1])
 
         # Step 2
-        publication = publish(self.cfg, publisher, repo)
+        publication1 = publication(self.cfg, DOCKER_PUBLICATION_PATH, repo)
```

Comment:
Pulp-Smash now has a `task_handler` it was added recently in https://github.com/PulpQE/pulp-smash/pull/1184

Now you don't need that extra `publication` function and can simply do

```python
publication1 = self.client.using_handler(api.task_handler).post(
    DOCKER_PUBLICATION_PATH, 
    data={"repository": repo["_href"]}
)
```
The above `post` response will be passed to `api.task_handler` which will do exactly what your `publication` function is doing see:  https://github.com/PulpQE/pulp-smash/blob/master/pulp_smash/api.py#L205-L268